processing/src/main/java/org/apache/carbondata/processing/loading/steps/DataConverterProcessorStepImpl.java

```diff
@@ -100,10 +103,17 @@ public void initialize() throws IOException {
    * @return processed row.
    */
   protected CarbonRowBatch processRowBatch(CarbonRowBatch rowBatch, RowConverter localConverter) {
+    String badRecordsAction = (String) configuration.getDataLoadProperty(BAD_RECORDS_LOGGER_ACTION);
     while (rowBatch.hasNext()) {
       CarbonRow convertRow = localConverter.convert(rowBatch.next());
-      rowBatch.setPreviousRow(convertRow);
+      if (convertRow == null && (badRecordsAction.equals(LoggerAction.IGNORE.toString()) ||
+          badRecordsAction.equals(LoggerAction.REDIRECT.toString()))) {
```

Comment:
Please remove check (badRecordsAction.equals(LoggerAction.IGNORE.toString()) ||
           badRecordsAction.equals(LoggerAction.REDIRECT.toString()) 
BadRecordsAction check is not required, only null check over convertRow is enough.
The convert method return's null only in case of bad_records_action is either IGNORE or REDIRECT.

Reply:
@geetikagupta16 and @QiangCai 
In case of bucketing and Stream Record writer also the same exist. 
Please correct there as well.
For your reference:
1.  For  bucketing.
 org/apache/carbondata/processing/loading/steps/DataConverterProcessorWithBucketingStepImpl.java:128
2. For Stream Record Writer 
org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java:191